### PR TITLE
Update ApplicationInsightsSDK, DiagnosticSource and minor application…

### PR DESCRIPTION
**Pull request by cijothomas**
…insights exporter fixes

### CODE REVIEW

File: src/OpenTelemetry.Exporter.ApplicationInsights/ApplicationInsightsTraceExporter.cs [hunk 1]
```diff
@@ -38,6 +39,7 @@ public class ApplicationInsightsTraceExporter : SpanExporter, IDisposable
         public ApplicationInsightsTraceExporter(TelemetryConfiguration telemetryConfiguration)
         {
             this.telemetryClient = new TelemetryClient(telemetryConfiguration);
+            this.telemetryClient.Context.GetInternalContext().SdkVersion = "ot:" + GetAssemblyVersion();
```
Comment (lmolkova) [inline src/OpenTelemetry.Exporter.ApplicationInsights/ApplicationInsightsTraceExporter.cs]: there will be more changes coming here (for OpenCensus we also had language here) https://github.com/microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/SDK-VERSIONS.md#sdk-names

It's fine to keep it, but we should understand our strategy here and whether we want to support it through current schema or do nicer less verbose things through resource API

File: test/OpenTelemetry.Exporter.LightStep.Tests/OpenTelemetry.Exporter.LightStep.Tests.csproj [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
     <ItemGroup>
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.0.1" />
-        <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.6.0-preview7.19362.9" />
+        <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
```
Comment (lmolkova) [inline test/OpenTelemetry.Exporter.LightStep.Tests/OpenTelemetry.Exporter.LightStep.Tests.csproj]: this reference should be removed at all - it is coming through OpenTelemetry.Abstractions

File: src/OpenTelemetry.Exporter.ApplicationInsights/ApplicationInsightsTraceExporter.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.ApplicationInsights/ApplicationInsightsTraceExporter.cs]: yes i understand. i added this as a mere starting point to help me distinguish which telemetry was coming from OT exporter.

**Comment by cijothomas** [comment]
#please-merge

**Pull request merged**